File: app/src/main/java/com/backbase/citysearch/models/City.java
package com.backbase.citysearch.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class City implements Comparable<City> {

    @SerializedName("country")
    @Expose
    private String country;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("_id")
    @Expose
    private long id;
    @SerializedName("coord")
    @Expose
    private Coord coord;

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public Coord getCoord() {
        return coord;
    }

    public void setCoord(Coord coord) {
        this.coord = coord;
    }

    @Override
    public int compareTo(City o) {
        int nameCompareValue = this.name.compareTo(o.name);
        if (nameCompareValue == 0) {
            return this.country.compareTo(o.country);
        } else {
            return nameCompareValue;
        }
    }
}

File: app/src/main/java/com/backbase/citysearch/models/City.kt
package com.backbase.citysearch.models

import com.backbase.citysearch.models.City
import com.google.gson.annotations.SerializedName
import com.google.gson.annotations.Expose
import com.backbase.citysearch.models.Coord

class City : Comparable<City> {
    @SerializedName("country")
    @Expose
    var country: String? = null

    @SerializedName("name")
    @Expose
    var name: String? = null

    @SerializedName("_id")
    @Expose
    var id: Long = 0

    @SerializedName("coord")
    @Expose
    var coord: Coord? = null
    override fun compareTo(o: City): Int {
        val nameCompareValue = name!!.compareTo(o.name!!)
        return if (nameCompareValue == 0) {
            country!!.compareTo(o.country!!)
        } else {
            nameCompareValue
        }
    }
}